tdrs-backend/tdpservice/parsers/tdr_parser.py

```diff
@@ -358,3 +363,92 @@ def _generate_trailer_errors(self, trailer_errors):
         if settings.GENERATE_TRAILER_ERRORS:
             self.unsaved_parser_errors.update({"trailer": trailer_errors})
             self.num_errors += len(trailer_errors)
+
+    def _generate_funded_ssn_errors(self, t2_schema, t2_records, t1_schema):
+        """Inner function to validate and generate errors."""
+        current_index = settings.BULK_CREATE_BATCH_SIZE
+        prev_index = 0
+        num_records = t2_records.count()
+        validator = category2.ssnAllOf(
+            category2.isNumber(),
+            category2.intHasLength(9),
+            category2.valueNotAt(slice(0, 3), "000"),
+            category2.valueNotAt(slice(0, 3), "666"),
+            category2.valueNotAt(slice(3, 5), "00"),
+            category2.valueNotAt(slice(5, 9), "0000"),
+            error_message="Federally funded recipients must have a valid SSN.",  # TODO: need input on this
+        )
+        t1_field_names = ("FUNDING_STREAM",)
+        t2_field_names = ("FAMILY_AFFILIATION", "SSN")
+        t1_fields = [t1_schema.get_field_by_name(name) for name in t1_field_names]
+        t2_fields = [t2_schema.get_field_by_name(name) for name in t2_field_names]
+        fields = t1_fields + t2_fields
+        while True:
```

Comment:
do we need `while True` since we aren't deleting records from the queryset as we process?